keras_core/layers/preprocessing/category_encoding.py

```diff
@@ -84,21 +84,23 @@ class CategoryEncoding(Layer):
     """
 
     def __init__(self, num_tokens=None, output_mode="multi_hot", **kwargs):
-        if not tf.available:
-            raise ImportError(
-                "Layer CategoryEncoding requires TensorFlow. "
-                "Install it via `pip install tensorflow`."
-            )
-
         super().__init__(**kwargs)
 
         # Support deprecated names for output_modes.
         if output_mode == "binary":
             output_mode = "multi_hot"
+
         # 'output_mode' must be one of ("count", "one_hot", "multi_hot")
         if output_mode not in ("count", "one_hot", "multi_hot"):
             raise ValueError(f"Unknown arg for output_mode: {output_mode}")
 
+        if output_mode == 'multi_hot':
+            self.output_function = backend.nn.multi_hot
```

Comment:
There is an additional subtlety here which is that the layer should be compatible with tf.data. You can achieve this by subclassing `TFDataLayer` and accessing the backend as `self.backend`. See an example in e.g. `center_crop.py`.

Reply:
I changed the base class to `TFDataLayer` and used `self.backend` instead of `backend`.